cluster/saltbase/salt/kubelet/default

```diff
@@ -151,6 +151,8 @@
   {% set network_plugin = "--network-plugin=opencontrail" %}
 {% elif pillar.get('network_provider', '').lower() == 'cni' %}
   {% set network_plugin = "--network-plugin=cni --network-plugin-dir=/etc/cni/net.d/" %}
+{%elif pillar.get('policy_provider', '').lower() == 'calico' and grains['roles'][0] != 'kubernetes-master' -%}
```

Comment:
Just to make sure we're on the same page and nothing is getting lost in translation -

As it stands, if you switch this on, kubenet won't get called, but in terms of user-facing behavior it doesn't really make a difference.  This will still run on top of any network kubenet would, including GCE (and in this PR can only be enabled on GCE).

The Calico CNI plugin doesn't use a bridge, so it doesn't need all of the config that kubenet sets up (cbr0, sysctls, promiscuous mode, haripin) in order to work.  It uses the host-local IPAM driver in the same way kubenet does.

As far as I can tell, the only thing kubenet does that wouldn't be supported here is all the bandwidth shaping stuff, which I think is OK for for now.  For now users can use --network-plugin=kubenet if they're interested in tc, and --network-plugin=cni if they're interested in policy, and we can work on how to merge those down the road.